packages/cli-hydrogen/src/cli/services/build/check-lockfile.ts

```diff
@@ -0,0 +1,94 @@
+import {file, git, path} from '@shopify/cli-kit'
+import {renderWarning} from '@shopify/cli-kit/node/ui'
+import {lockfiles} from '@shopify/cli-kit/node/node-package-manager'
+import type {Lockfile} from '@shopify/cli-kit/node/node-package-manager'
+
+function missingLockfileWarning() {
+  renderWarning({
+    headline: 'No lockfile found',
+    body:
+      `If you don’t commit a lockfile, then your app might install the wrong ` +
+      `package versions when deploying. To avoid versioning issues, generate a ` +
+      `new lockfile and commit it to your repository.`,
+    nextSteps: [
+      [
+        'Generate a lockfile. Run',
+        {
+          command: 'npm|yarn|pnpm install',
+        },
+      ],
+      'Commit the new file to your repository',
+    ],
+  })
+}
+
+function tooManyLockfilesWarning(lockfiles: Lockfile[]) {
+  const packageManagers = {
+    'yarn.lock': 'yarn',
+    'package-lock.json': 'npm',
+    'pnpm-lock.yaml': 'pnpm',
+  }
+
+  const lockfileList = lockfiles.map((lockfile) => {
+    return `• ${lockfile} (created by ${packageManagers[lockfile]})`
+  })
+
+  renderWarning({
+    headline: 'Multiple lockfiles found',
+    body: [
+      `Your project contains more than one lockfile. This can cause version ` +
+        `conflicts when installing and deploying your app. The following ` +
+        `lockfiles were detected:\n\n`,
+      lockfileList.join('\n '),
+    ],
+    nextSteps: ['Delete any unneeded lockfiles', 'Commit the change to your repository'],
+  })
+}
+
+function lockfileIgnoredWarning(lockfile: string) {
+  renderWarning({
+    headline: 'Lockfile ignored by Git',
+    body:
+      `Your project’s lockfile isn’t being tracked by Git. If you don’t commit ` +
+      `a lockfile, then your app might install the wrong package versions when ` +
+      `deploying.`,
+    nextSteps: [
+      `In your project’s .gitignore file, delete any references to ${lockfile}`,
+      'Commit the change to your repository',
+    ],
+  })
+}
+
+type LockFileStatus = 'missing' | 'tooMany' | 'ignored' | 'ok'
+
+export async function checkForMissingLockfile(directory: string): Promise<LockFileStatus> {
+  const availableLockFiles = await lockfiles.reduce(async (acc, lockFileName) => {
+    const lockfilePath = path.resolve(directory, lockFileName)
+    if (await file.exists(lockfilePath)) {
+      return (await acc).concat(lockFileName)
+    } else {
+      return acc
+    }
+  }, Promise.resolve([] as Lockfile[]))
+
+  if (!availableLockFiles.length) {
+    missingLockfileWarning()
+    return 'missing'
+  }
+
+  if (availableLockFiles.length > 1) {
+    tooManyLockfilesWarning(availableLockFiles)
```

Comment:
Slight inconsistency in casing in this file: LockFile vs Lockfile. No real preference, just that it is one or the other.